File: app/src/main/java/org/app/sharing/dto/PagedNews.java
package org.app.sharing.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.app.note.dto.RoomNoteDto;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PagedNews {
    private List<RoomNoteDto> content;
    private int  page;
    private int  size;
    private long totalElements;
    private int  totalPages;
    private boolean last;
}

File: app/src/main/java/com/coursework/pleasantroutineui/pleasantroutineui/domain/discovery/PagedNews.kt
package com.coursework.pleasantroutineui.domain.discovery

import com.coursework.pleasantroutineui.domain.Note

data class PagedNews(
    val content: List<Note>,
    val page: Int,
    val size: Int,
    val totalElements: Long,
    val totalPages: Int,
    val last: Boolean
)